contracts/Proxy/TimeOwnedUpgradeabilityProxy.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity ^0.5.13;
+
+import { OwnedUpgradeabilityProxy } from "./OwnedUpgradeabilityProxy.sol";
+
+/**
+ * @title TimeOwnedUpgradeabilityProxy
+ * @dev This contract combines an upgradeability proxy with 
+ * basic authorization control functionalities
+ *
+ * This contract allows us to specify a time at which the proxy can no longer
+ * be upgraded
+ */
+contract TimeOwnedUpgradeabilityProxy is OwnedUpgradeabilityProxy {
+
+    uint256 expiration;
+    uint256 constant MAX_UINT = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+
+    /**
+    * @dev the constructor sets the original owner of the contract to the sender account.
+    */
+    constructor() public {
+        _setUpgradeabilityOwner(msg.sender);
+        expiration = MAX_UINT;
+    }
+
+    /**
+     *
+    */
+    function setExpiration(uint256 _newTimestamp) external onlyProxyOwner {
+        expiration = _newTimestamp;
```

Comment:
This should also have and if() else or require(block.timestamp < expiration)

Reply:
Because once you're past the expiration you shouldn't be able to change the expiration